src/meta-srv/src/procedure/region_migration/manager.rs

```diff
@@ -275,6 +277,15 @@ impl RegionMigrationManager {
             }
         );
 
+        if task.from_peer.addr.is_empty() {
```

Comment:
How can `from_peer`'s addr be empty?

Reply:
It occurs when the datanode never connects to the new leader.
```rust
        let from_peer = self
            .peer_lookup
            .datanode(from_peer_id)
            .await
            .ok()
            .flatten()
            .unwrap_or_else(|| Peer::empty(from_peer_id));
```